client/asset/eth/eth.go

```diff
@@ -142,11 +142,13 @@ func (d *Driver) DecodeCoinID(coinID []byte) (string, error) {
 		}
 		return id.String(), nil
 	}
-	id, err := decodeFundingCoinID(coinID)
-	if err != nil {
-		return invalid, err
+
+	id := string(coinID)
+	if len(id) != 42 || id[0:2] != "0x" {
+		return invalid, fmt.Errorf("coin is not byte encoded utf-8 string of ETH address: %x", coinID)
```

Comment:
Even now it's difficult to keep track of the coin ID variations for ETH.  Let me see if I can summarize.

1. a tx hash
2. just an address/account, utf-8 encoded
3. funding coin / recover encoding, the bytes of account || amount

So, the `eth.fundingCoin` type can give both the utf-8 encoded account via `ID()` or a funding coin encoding via `RecoveryID()`.   The plain utf-8 encoded address/account is used when placing orders for giving server an account without the amount.

Most places with server interaction for the actual swap will use the `eth.coin` type with a tx hash encoded.

What ends up in a `core.Order` returned by `(*Core).Order` (used by the http endoint for this page, I think) comes from `coreOrderFromTrade`, which gets it from `ord.Trade().Coins[]`.  These guys are set in `prepareTrackedTrade` from calling the `eth.(*fundingCoin).ID` method that returns the utf-8 encoded account as described above.
Finally, this `Driver` method you are fixing is used by `core.NewCoin` in `coreOrderFromTrade`?

:sweat: 

So I'm wondering what else uses this driver method and if it  needs to be designed to work with the recovery encoding (3) as well as the tx id and account utf-8.

Reply:
Yeah you're right we need all three, my bad.

Reply:
I wonder if encoding a bit in the eth coin id would make it more organized, jk

Reply:
Lololol